pkg/quarterdeck/auth.go

```diff
@@ -101,17 +103,19 @@ func (s *Server) Register(c *gin.Context) {
 			return
 		}
 
-		log.Error().Err(err).Msg("could not insert user into database during registration")
+		c.Error(err)
 		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not process registration"))
 		return
 	}
 
 	// Verification emails should happen asynchronously because sending emails can be
 	// slow and waiting for SendGrid to send the email could cause the request to time
 	// out even though the user was successfully created.
+	hub := sentrygin.GetHubFromContext(c).Clone()
 	s.tasks.Queue(tasks.TaskFunc(func(ctx context.Context) {
 		if err := s.SendVerificationEmail(user); err != nil {
 			log.Error().Err(err).Str("user_id", user.ID.String()).Msg("could not send verification email to user")
+			hub.CaptureException(err)
```

Comment:
This is my attempt to ensure concurrency safety in sentry, RE the following guide:

https://docs.sentry.io/platforms/go/guides/gin/concurrency/